backend/Dockerfile

```diff
@@ -31,6 +31,10 @@ RUN wget https://bootstrap.pypa.io/get-pip.py && python3 get-pip.py
 # pin avro==1.9.1 because installing just tfx brings avro 1.9.2, which breaks the build
 RUN python3 -m pip install avro-python3==1.9.1 tfx==0.21.0rc0
 
+# Downloading Argo CLI so that the samples are validated
+ADD  https://github.com/argoproj/argo/releases/download/v2.3.0/argo-linux-amd64 /usr/local/bin/argo
+RUN chmod +x /usr/local/bin/argo
+
```

Comment:
I may didn't get the meaning in the SDK side PR.
I think it should be validated in SDK/Client.

To learn more, here are few questions:
Why put it to backend and via CLI?
When/where the CLI got called? Inside API Server after get a request? It's not a typical pattern to launch a subprocess after get a API request, right?

Reply:
> I think it should be validated in SDK/Client.

It's already there.

>When/where the CLI got called? Inside API Server after get a request? It's not a typical pattern to launch a subprocess after get a API request, right?

It's not used by the ApiServer main container. It's used by the sample builder container that builds preloaded samples. We want to make sure that the preloaded samples are compiled correctly.